services/src/northbound/src/main/java/org/openkilda/northbound/controller/SwitchController.java

```diff
@@ -275,4 +277,36 @@ public PortDto configurePort(
         LOGGER.info("Port Configuration '{}' request for port {} of switch {}", portConfig, portNo, switchId);
         return switchService.configurePort(switchId, portNo, portConfig);
     }
+
+    /**
+     * Get switch port descriptions.
+     *
+     * @param switchId the switch id.
+     * @return switch port descriptions.
+     */
+    @ApiOperation(value = "Get switch ports description from the switch", response = SwitchPortsDescription.class)
+    @GetMapping(value = "/switches/{switch-id}/port_desc",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public SwitchPortsDescription getSwitchPortsDescription(
+            @PathVariable("switch-id") SwitchId switchId) {
+        return switchService.getSwitchPortsDescription(switchId);
+    }
+
+    /**
+     * Get port description of the switch.
+     *
+     * @param switchId the switch id.
+     * @param port the port of the switch.
+     * @return port description.
+     */
+    @ApiOperation(value = "Get port description from the switch", response = PortDescription.class)
+    @GetMapping(value = "/switches/{switch-id}/port_desc/{port}",
```

Comment:
This URI doesn't follow RESTful API style: there's no such entity like "port_desc" to represent it as a resource. Reconsider to use just ".../ports/{port}"